src/test/java/org/fogbowcloud/manager/core/models/linkedlist/SynchronizedDoublyLinkedListTest.java

```diff
@@ -0,0 +1,242 @@
+		
+		this.list.resetPointer();
+		assertEquals(orderOne, this.list.getNext());
+		assertEquals(orderTwo, this.list.getNext());
+		assertNull(this.list.getNext());
+	}
+
+	@Test
+	public void testRemoveNullOrder() {
+		assertNull(this.list.getNext());
+
+		Order orderNull = null;
+		try {
+			this.list.removeItem(orderNull);
+			fail("Null order should not be removed.");
+		} catch (IllegalArgumentException e){
+			assertEquals("Attempting to add a null order.", e.getMessage());
+		}
+	}
+	
+	@Test
+	public void testFindNodeToRemove() {
+		assertNull(this.list.getNext());
+
+		Order orderOne = createOrder("one");
+		Order orderTwo = createOrder("two");
+		Order orderThree = createOrder("three");
+		Order orderFour = createOrder("four");
+		this.list.addItem(orderOne);
+		this.list.addItem(orderTwo);
+		this.list.addItem(orderThree);
+		this.list.addItem(orderFour);
+		
+		Node nodeToRemove = this.list.findNodeToRemove(orderTwo);
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderTwo, nodeToRemove.getOrder());
+		assertEquals(orderOne, nodeToRemove.getPrevious().getOrder());
+		assertEquals(orderThree, nodeToRemove.getNext().getOrder());
+		assertEquals(orderFour, this.list.getTail().getOrder());
+	}
+	
+	@Test
+	public void testRemoveItemOnHead() {
+		Order orderOne = createOrder("one"); 
+		Order orderTwo = createOrder("two"); 
+		Order orderThree = createOrder("three");
+		Order orderFour = createOrder("four");
+		
+		this.list.addItem(orderOne);
+		this.list.addItem(orderTwo);
+		this.list.addItem(orderThree);
+		this.list.addItem(orderFour);
+
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderFour, this.list.getTail().getOrder());
+		
+		assertEquals(orderOne, this.list.getCurrent().getOrder());
+		this.list.removeItem(orderOne);
+		assertEquals(orderTwo, this.list.getCurrent().getOrder());
+		assertEquals(orderTwo, this.list.getHead().getOrder());
+		assertNull(this.list.getHead().getPrevious());
+		assertEquals(orderFour, this.list.getTail().getOrder());
+		assertEquals(orderTwo, this.list.getNext());
+		assertEquals(orderThree, this.list.getNext());
+		assertEquals(orderFour, this.list.getNext());
+		assertNull(this.list.getNext());
+	}
+	
+	@Test
+	public void testRemoveItemOnTail() {
+		Order orderOne = createOrder("one"); 
+		Order orderTwo = createOrder("two"); 
+		Order orderThree = createOrder("three");
+
+		this.list.addItem(orderOne);
+		this.list.addItem(orderTwo);
+		this.list.addItem(orderThree);
+
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderThree, this.list.getTail().getOrder());
+		
+		this.list.removeItem(orderThree);
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderTwo, this.list.getTail().getOrder());
+		assertEquals(orderOne, this.list.getNext());
+		assertEquals(orderTwo, this.list.getNext());
+		assertNull(this.list.getNext());
+	}
+	
+	@Test
+	public void testRemoveItemOneElementOnList() {
```

Comment:
To be consistent with the other tests, include an assert to guarantee that the list starts empty.